File: libs/DependencyDownloader/src/main/java/org/wdt/utils/dependency/DependencyDownload.java
package org.wdt.utils.dependency;

import java.io.File;
import java.io.IOException;
import java.net.MalformedURLException;
import java.net.URL;

public class DependencyDownload extends DefaultDependency {
  private File DownloadPath;
  private String DefaultUrl = "https://repo1.maven.org/maven2/";

  public DependencyDownload(String lib) {
    super(lib);
  }

  public DependencyDownload(String groupId, String artifactId, String version) {
    super(groupId, artifactId, version);
  }

  public String getLibraryFilePath() throws IOException {
    return getLibraryFile().getCanonicalPath();
  }

  public File getLibraryFile() {
    return new File(DownloadPath, formJar());
  }


  public String getDefaultUrl() {
    return DefaultUrl;
  }

  public void setDefaultUrl(String defaultUrl) {
    DefaultUrl = defaultUrl;
  }

  public File getDownloadPath() {
    return DownloadPath;
  }

  public void setDownloadPath(File path) {
    this.DownloadPath = path;
  }


  public URL getLibraryUrl() throws MalformedURLException {
    return new URL(DefaultUrl + formJar());
  }

  public LibraryDependencyDownload getLibraryDependencyDownload() {
    LibraryDependencyDownload libraryDependencyDownload = GetLibraryDependencyDownload();
    libraryDependencyDownload.setDefaultUrl(DefaultUrl);
    return libraryDependencyDownload;
  }
}

File: WdtcCore/src/main/kotlin/org/wdt/wdtc/core/utils/dependency/DependencyDownload.kt
package org.wdt.wdtc.core.utils.dependency

import java.io.File
import java.io.IOException
import java.net.MalformedURLException
import java.net.URL

class DependencyDownload : DefaultDependency {
  var downloadPath: File? = null
  var defaultUrl = "https://repo1.maven.org/maven2/"

  constructor(lib: String) : super(lib)
  constructor(groupId: String, artifactId: String, version: String) : super(groupId, artifactId, version)

  @get:Throws(IOException::class)
  val libraryFilePath: String
    get() = libraryFile.getCanonicalPath()
  val libraryFile: File
    get() = File(downloadPath, formJar())

  @get:Throws(MalformedURLException::class)
  val libraryUrl: URL
    get() = URL(defaultUrl + formJar())
}